src/lib/source-handlers/bitbucket-server/list-projects.ts

```diff
@@ -0,0 +1,87 @@
+import * as needle from 'needle';
+import * as debugLib from 'debug';
+import Bottleneck from 'bottleneck';
+import { BitbucketServerProjectData } from './types';
+import { getBitbucketServerToken } from './get-bitbucket-server-token';
+
+const debug = debugLib('snyk:bitbucket-server');
+
+const limiter = new Bottleneck({
+  reservoir: 1000, // initial value
+  reservoirRefreshAmount: 1000,
+  reservoirRefreshInterval: 3600 * 1000,
+  maxConcurrent: 1,
+  minTime: 1000,
+});
+
+limiter.on('failed', async (error, jobInfo) => {
+  const id = jobInfo.options.id;
+  debug(`Job ${id} failed: ${error}`);
+  if (jobInfo.retryCount === 0) {
+    // Here we only retry once
+    debug(`Retrying job ${id} in 25ms!`);
+    return 25;
+  }
+});
+
+export const fetchAllProjects = async (
+  url: string,
+  token: string,
+): Promise<unknown[]> => {
+  let isLastPage = false;
+  let values: unknown[] = [];
+  let pageCount = 1;
+  let start = 0;
+  const limit = 100;
+  while (!isLastPage) {
+    debug(`Fetching page ${pageCount} for projects\n`);
+    const response = await limiter.schedule(() =>
```

Comment:
For readability only, could you destructure everything you need here, rather than access it later as `response.something`, like so:

```
const { statusCode, body } = await limiter.schedule


```

Reply:
done